### PR TITLE
Refactor rx

### LINKED ISSUE #11555: @types is still broken for rx-lite
**TypeScript Version:**  next (2.1.0-dev.20161012)

The problems I raised in #9725 are still present. TypeScript 2 still refuses to recognise the "rx-lite" module declared by "@types/rx".

**Code**

```
npm init
npm install typescript@next rx @types/rx --save
```

**tsconfig.json**

``` ts
{
  "compilerOptions": {
    "module": "es6",
    "target": "es5",
    "moduleResolution": "node",
    "types": [
      "rx"
    ]
  },
  "files": [
    "import-rx-lite.ts"
  ],
  "exclude": [
      "node_modules"
  ]
}
```

**import-rx-lite.ts**

``` ts
import * as Rx from 'rx-lite';
```

```
./node-modules/.bin/tsc
```

**Expected behavior:**

It should work.

**Actual behavior:** 

```
import-rx-lite.ts(1,16): error TS2307: Cannot find module 'rx-lite'.
```

**Example Project**

https://github.com/jonrimmer/ts-types-bug

### PR DESCRIPTION
**Pull request by mhegazy**
This refactors the rx declaration into different packages. the packages should map directly to the rx npm packages under https://www.npmjs.com/~mattpodwysocki

I did not fix lint errors, there were too many of them :). I have also not switched the declarations into modules, i can do that in a separate PR.

Fixes https://github.com/Microsoft/TypeScript/issues/11555

//cc @andy-ms

### CODE REVIEW

File: rx-dom/index.d.ts [hunk 1]
```diff
@@ -137,6 +137,6 @@ declare module Rx.DOM {
     }
 }
 
-declare module "rx.DOM" {
+declare module "rx-dom" {
```
Comment (DanielRosenwasser) [inline rx-dom/index.d.ts]: Did these things ever work?

**Comment by DanielRosenwasser** [comment]
As far as I can tell this looks like a decent change. I might opt to fix up the indentation to just spaces so the comments align appropriately, but that's me being anal.

Also, I don't really know whether it's appropriate to switch `strictNullChecks` on, but I guess it doesn't hurt for testing purposes.

File: rx-dom/index.d.ts (hunk 1 above)
Comment (mhegazy) [inline rx-dom/index.d.ts]: i have no idea. could not find any documentation for the ones with the `.` in the name.

File: rx-core-binding/index.d.ts [hunk 2]
```diff
@@ -1,9 +1,34 @@
...
+// Type definitions for rx-core-binding 4.0
+// Project: https://github.com/Reactive-Extensions/RxJS
+// Definitions by: Carl de Billy <http://carl.debilly.net/>, Igor Oleinikov <https://github.com/Igorbek>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-///<reference path="rx-lite.d.ts"/>
+///<reference types="rx-core"/>
 
 declare namespace Rx {
+	interface ISubject<T> extends Observable<T>, Observer<T>, IDisposable {
```
Comment (andy-ms) [inline rx-core-binding/index.d.ts]: Mixed tabs and spaces.

**Review by ghost — COMMENTED** [review]
Otherwise looks good. Basically what @DanielRosenwasser said.

**Pull request merged**